### PR TITLE
Added rds as backing service

### PR DESCRIPTION
**Pull request by goruha**
## What
* Added RDS as backing service

## Why
* To make it useful for some projects

### CODE REVIEW

File: aws/backing-services/rds.tf [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+variable "RDS_CLUSTER_ENABLED" {
+  type        = "string"
+  default     = "true"
+  description = "Set to false to prevent the module from creating any resources"
+}
+
+variable "RDS_SNAPSHOT" {
+  type        = "string"
+  default     = ""
+  description = "Restore snapshots"
```
Comment (aknysh) [inline aws/backing-services/rds.tf]: Set to a snapshot ID to restore from snapshot

File: aws/backing-services/rds.tf [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+variable "RDS_PARAMETER_GROUP_NAME" {
+  type        = "string"
+  default     = ""
+  description = "Existed paramater group name to use"
```
Comment (aknysh) [inline aws/backing-services/rds.tf]: Existing parameter group name to use

File: aws/backing-services/rds.tf [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  engine                      = "${var.RDS_ENGINE}"
+  engine_version              = "${var.RDS_ENGINE_VERSION}"
+  instance_class              = "${var.RDS_INSTANCE_TYPE}"
+  db_parameter_group          = "${var.RDS_DB_PARAMETER_GROUP}"
+  parameter_group_name        = "${var.RDS_PARAMETER_GROUP_NAME}"
+  publicly_accessible         = "false"
+  subnet_ids                  = ["${module.subnets.private_subnet_ids}"]
+  vpc_id                      = "${module.vpc.vpc_id}"
+  snapshot_identifier         = "${var.RDS_SNAPSHOT}"
+  auto_minor_version_upgrade  = "false"
```
Comment (aknysh) [inline aws/backing-services/rds.tf]: parameterize?

**Comment by goruha** [comment]
> see comments.
> also, let's rename the file to aws/backing-services/rds-mysql.tf since it's just for MySQL and we could > have other RDS databases

No. The same `rds.tf` can be applied to any types of db except aurora. 
Mysql is just default one

File: aws/backing-services/rds.tf [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+  default     = "22:00-03:00"
+  description = "When AWS can perform DB snapshots, can't overlap with maintenance window"
+}
+
+module "rds" {
+  source                      = "git::https://github.com/cloudposse/terraform-aws-rds.git?ref=tags/0.4.1"
+  enabled                     = "${var.RDS_ENABLED}"
+  namespace                   = "${var.namespace}"
+  stage                       = "${var.stage}"
+  name                        = "rds"
```
Comment (aknysh) [inline aws/backing-services/rds.tf]: let's add and use variable `name` here

**Pull request merged**